### ISSUE TITLE
LoadError: libruby.so.3.2: cannot open shared object file: No such file or directory - /usr/local/bundle/gems/stringio-3.1.1/lib/stringio.so

### ISSUE THREAD

**Issue by excelsior** [rubygems/rubygems#8099]
I'm trying to build a Docker image for a Rails app using Ruby 3.3.5 and Bundler 2.5.16. Running `docker compose up --build` fails with this message:

```
LoadError: libruby.so.3.2: cannot open shared object file: No such file or directory - /usr/local/bundle/gems/stringio-3.1.1/lib/stringio.so
  <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          <internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          /usr/local/lib/ruby/3.3.0/rubygems/remote_fetcher.rb:78:in `initialize'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:211:in `new'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:211:in `gem_remote_fetcher'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:232:in `block in fetchers'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:232:in `map'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:232:in `fetchers'
          /usr/local/lib/ruby/3.3.0/bundler/fetcher.rb:205:in `api_fetcher?'
          /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:400:in `select'
          /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:400:in `api_fetchers'
          /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:405:in `block in remote_specs'
          /usr/local/lib/ruby/3.3.0/bundler/index.rb:9:in `build'
          /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:404:in `remote_specs'
          /usr/local/lib/ruby/3.3.0/bundler/source/rubygems.rb:138:in `specs'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:41:in `block in setup_solver'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:247:in `all_versions_for'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:54:in `block in setup_solver'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:351:in `filtered_versions_for'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:58:in `block in setup_solver'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:428:in `select_sorted_versions'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:407:in `block in prepare_dependencies'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:401:in `each'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:401:in `map'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:401:in `prepare_dependencies'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:63:in `setup_solver'
          /usr/local/lib/ruby/3.3.0/bundler/resolver.rb:28:in `start'
          /usr/local/lib/ruby/3.3.0/bundler/definition.rb:633:in `start_resolution'
          /usr/local/lib/ruby/3.3.0/bundler/definition.rb:305:in `resolve'
          /usr/local/lib/ruby/3.3.0/bundler/definition.rb:179:in `resolve_remotely!'
          /usr/local/lib/ruby/3.3.0/bundler/installer.rb:256:in `resolve_if_needed'
          /usr/local/lib/ruby/3.3.0/bundler/installer.rb:80:in `block in run'
          /usr/local/lib/ruby/3.3.0/bundler/process_lock.rb:12:in `block in lock'
          /usr/local/lib/ruby/3.3.0/bundler/process_lock.rb:9:in `open'
          /usr/local/lib/ruby/3.3.0/bundler/process_lock.rb:9:in `lock'
          /usr/local/lib/ruby/3.3.0/bundler/installer.rb:71:in `run'
          /usr/local/lib/ruby/3.3.0/bundler/installer.rb:23:in `install'
          /usr/local/lib/ruby/3.3.0/bundler/cli/install.rb:63:in `run'
          /usr/local/lib/ruby/3.3.0/bundler/cli.rb:247:in `block in install'
          /usr/local/lib/ruby/3.3.0/bundler/settings.rb:158:in `temporary'
          /usr/local/lib/ruby/3.3.0/bundler/cli.rb:246:in `install'
          /usr/local/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
          /usr/local/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
          /usr/local/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
          /usr/local/lib/ruby/3.3.0/bundler/cli.rb:35:in `dispatch'
          /usr/local/lib/ruby/3.3.0/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
          /usr/local/lib/ruby/3.3.0/bundler/cli.rb:29:in `start'
          /usr/local/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:28:in `block in <top (required)>'
          /usr/local/lib/ruby/3.3.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
          /usr/local/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:20:in `<top (required)>'
          /usr/local/bin/bundle:25:in `load'
          /usr/local/bin/bundle:25:in `<main>'

```

## Environment

```
Bundler       2.5.16
  Platforms   ruby, aarch64-linux
Ruby          3.3.5p100 (2024-09-03 revision ef084cc8f4958c1b6e4ead99136631bef6d8ddba) [aarch64-linux]
  Full Path   /usr/local/bin/ruby
  Config Dir  /usr/local/etc
RubyGems      3.5.16
  Gem Home    /usr/local/bundle
  Gem Path    /root/.local/share/gem/ruby/3.3.0:/usr/local/lib/ruby/gems/3.3.0:/usr/local/bundle
  User Home   /root
  User Path   /root/.local/share/gem/ruby/3.3.0
  Bin Dir     /usr/local/bundle/bin
OpenSSL       
  Compiled    OpenSSL 3.0.14 4 Jun 2024
  Loaded      OpenSSL 3.0.14 4 Jun 2024
  Cert File   /usr/lib/ssl/cert.pem
  Cert Dir    /usr/lib/ssl/certs
Tools         
  Git         2.39.5
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2024-10-03
Git SHA           unknown
Released Version  false
```

## Bundler settings

```
app_config
  Set via BUNDLE_APP_CONFIG: "/usr/local/bundle"
silence_root_warning
  Set via BUNDLE_SILENCE_ROOT_WARNING: true
```

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '3.3.5'

gem 'airbrake', '~> 13.0'
gem 'aws-sdk-s3', '~> 1.167'
gem 'bundler', '= 2.5.16'
gem 'doorkeeper', '~> 5.7'
gem 'doorkeeper-jwt', '~> 0.4'
gem 'elasticsearch', '= 7.9.0'
gem 'faraday', '~> 1.10'
gem 'input_sanitizer', '~> 0.6'
gem 'logstash-event', '~> 1.2'
gem 'parallel', '~> 1.26'
gem 'pg', '~> 1.5'
gem 'puma', '~> 6.4'
gem 'rails', '= 7.2.1'
gem 'rexml', '~> 3.3'
gem 'searchkick', '~> 5.4'
gem 'sidekiq', '~> 7.3'
gem 'sidekiq-failures', '~> 1.0'

group :development do
  gem 'listen', '~> 3.9'
  gem 'pry-rails', '~> 0.3'
end

group :development, :test do
  gem 'byebug', '~> 11.1'
  gem 'factory_bot_rails', '~> 6.4'
  gem 'faker', '~> 3.4'
end

group :test do
  gem 'minitest-spec-rails', '~> 7.3'
  gem 'webmock', '~> 3.24'
end
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (7.2.1)
      actionpack (= 7.2.1)
      activesupport (= 7.2.1)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
      zeitwerk (~> 2.6)
    actionmailbox (7.2.1)
      actionpack (= 7.2.1)
      activejob (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      mail (>= 2.8.0)
    actionmailer (7.2.1)
      actionpack (= 7.2.1)
      actionview (= 7.2.1)
      activejob (= 7.2.1)
      activesupport (= 7.2.1)
      mail (>= 2.8.0)
      rails-dom-testing (~> 2.2)
    actionpack (7.2.1)
      actionview (= 7.2.1)
      activesupport (= 7.2.1)
      nokogiri (>= 1.8.5)
      racc
      rack (>= 2.2.4, < 3.2)
      rack-session (>= 1.0.1)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.2)
      rails-html-sanitizer (~> 1.6)
      useragent (~> 0.16)
    actiontext (7.2.1)
      actionpack (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      globalid (>= 0.6.0)
      nokogiri (>= 1.8.5)
    actionview (7.2.1)
      activesupport (= 7.2.1)
      builder (~> 3.1)
      erubi (~> 1.11)
      rails-dom-testing (~> 2.2)
      rails-html-sanitizer (~> 1.6)
    activejob (7.2.1)
      activesupport (= 7.2.1)
      globalid (>= 0.3.6)
    activemodel (7.2.1)
      activesupport (= 7.2.1)
    activerecord (7.2.1)
      activemodel (= 7.2.1)
      activesupport (= 7.2.1)
      timeout (>= 0.4.0)
    activestorage (7.2.1)
      actionpack (= 7.2.1)
      activejob (= 7.2.1)
      activerecord (= 7.2.1)
      activesupport (= 7.2.1)
      marcel (~> 1.0)
    activesupport (7.2.1)
      base64
      bigdecimal
      concurrent-ruby (~> 1.0, >= 1.3.1)
      connection_pool (>= 2.2.5)
      drb
      i18n (>= 1.6, < 2)
      logger (>= 1.4.2)
      minitest (>= 5.1)
      securerandom (>= 0.3)
      tzinfo (~> 2.0, >= 2.0.5)
    addressable (2.8.7)
      public_suffix (>= 2.0.2, < 7.0)
    airbrake (13.0.4)
      airbrake-ruby (~> 6.0)
    airbrake-ruby (6.2.2)
      rbtree3 (~> 0.6)
    aws-eventstream (1.3.0)
    aws-partitions (1.983.0)
    aws-sdk-core (3.209.1)
      aws-eventstream (~> 1, >= 1.3.0)
      aws-partitions (~> 1, >= 1.651.0)
      aws-sigv4 (~> 1.9)
      jmespath (~> 1, >= 1.6.1)
    aws-sdk-kms (1.94.0)
      aws-sdk-core (~> 3, >= 3.207.0)
      aws-sigv4 (~> 1.5)
    aws-sdk-s3 (1.167.0)
      aws-sdk-core (~> 3, >= 3.207.0)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.5)
    aws-sigv4 (1.10.0)
      aws-eventstream (~> 1, >= 1.0.2)
    base64 (0.2.0)
    bigdecimal (3.1.8)
    builder (3.3.0)
    byebug (11.1.3)
    coderay (1.1.3)
    concurrent-ruby (1.3.4)
    connection_pool (2.4.1)
    crack (1.0.0)
      bigdecimal
      rexml
    crass (1.0.6)
    date (3.3.4)
    doorkeeper (5.7.1)
      railties (>= 5)
    doorkeeper-jwt (0.4.2)
      jwt (>= 2.1)
    drb (2.2.1)
    elasticsearch (7.9.0)
      elasticsearch-api (= 7.9.0)
      elasticsearch-transport (= 7.9.0)
    elasticsearch-api (7.9.0)
      multi_json
    elasticsearch-transport (7.9.0)
      faraday (~> 1)
      multi_json
    erubi (1.13.0)
    factory_bot (6.5.0)
      activesupport (>= 5.0.0)
    factory_bot_rails (6.4.3)
      factory_bot (~> 6.4)
      railties (>= 5.0.0)
    faker (3.4.2)
      i18n (>= 1.8.11, < 2)
    faraday (1.10.4)
      faraday-em_http (~> 1.0)
      faraday-em_synchrony (~> 1.0)
      faraday-excon (~> 1.1)
      faraday-httpclient (~> 1.0)
      faraday-multipart (~> 1.0)
      faraday-net_http (~> 1.0)
      faraday-net_http_persistent (~> 1.0)
      faraday-patron (~> 1.0)
      faraday-rack (~> 1.0)
      faraday-retry (~> 1.0)
      ruby2_keywords (>= 0.0.4)
    faraday-em_http (1.0.0)
    faraday-em_synchrony (1.0.0)
    faraday-excon (1.1.0)
    faraday-httpclient (1.0.1)
    faraday-multipart (1.0.4)
      multipart-post (~> 2)
    faraday-net_http (1.0.2)
    faraday-net_http_persistent (1.2.0)
    faraday-patron (1.0.0)
    faraday-rack (1.0.0)
    faraday-retry (1.0.3)
    ffi (1.17.0-aarch64-linux-gnu)
    ffi (1.17.0-arm64-darwin)
    ffi (1.17.0-x86_64-linux-gnu)
    globalid (1.2.1)
      activesupport (>= 6.1)
    hashdiff (1.1.1)
    hashie (5.0.0)
    i18n (1.14.6)
      concurrent-ruby (~> 1.0)
    input_sanitizer (0.6.0)
      activesupport (>= 3.0.0)
      method_struct (>= 0.2.2)
    io-console (0.7.2)
    irb (1.14.0)
      rdoc (>= 4.0.0)
      reline (>= 0.4.2)
    jmespath (1.6.2)
    jwt (2.9.0)
      base64
    listen (3.9.0)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    logger (1.6.1)
    logstash-event (1.2.02)
    loofah (2.22.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.12.0)
    mail (2.8.1)
      mini_mime (>= 0.1.1)
      net-imap
      net-pop
      net-smtp
    marcel (1.0.4)
    method_source (1.1.0)
    method_struct (0.3.0)
    mini_mime (1.1.5)
    minitest (5.25.1)
    minitest-spec-rails (7.3.0)
      minitest (>= 5.0)
      railties (>= 4.1)
    multi_json (1.15.0)
    multipart-post (2.4.1)
    net-imap (0.4.16)
      date
      net-protocol
    net-pop (0.1.2)
      net-protocol
    net-protocol (0.2.2)
      timeout
    net-smtp (0.5.0)
      net-protocol
    nio4r (2.7.3)
    nokogiri (1.16.7-aarch64-linux)
      racc (~> 1.4)
    nokogiri (1.16.7-arm64-darwin)
      racc (~> 1.4)
    nokogiri (1.16.7-x86_64-linux)
      racc (~> 1.4)
    parallel (1.26.3)
    pg (1.5.8)
    pry (0.14.2)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-rails (0.3.11)
      pry (>= 0.13.0)
    psych (5.1.2)
      stringio
    public_suffix (6.0.1)
    puma (6.4.3)
      nio4r (~> 2.0)
    racc (1.8.1)
    rack (3.1.7)
    rack-session (2.0.0)
      rack (>= 3.0.0)
    rack-test (2.1.0)
      rack (>= 1.3)
    rackup (2.1.0)
      rack (>= 3)
      webrick (~> 1.8)
    rails (7.2.1)
      actioncable (= 7.2.1)
      actionmailbox (= 7.2.1)
      actionmailer (= 7.2.1)
      actionpack (= 7.2.1)
      actiontext (= 7.2.1)
      actionview (= 7.2.1)
      activejob (= 7.2.1)
      activemodel (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      bundler (>= 1.15.0)
      railties (= 7.2.1)
    rails-dom-testing (2.2.0)
      activesupport (>= 5.0.0)
      minitest
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.6.0)
      loofah (~> 2.21)
      nokogiri (~> 1.14)
    railties (7.2.1)
      actionpack (= 7.2.1)
      activesupport (= 7.2.1)
      irb (~> 1.13)
      rackup (>= 1.0.0)
      rake (>= 12.2)
      thor (~> 1.0, >= 1.2.2)
      zeitwerk (~> 2.6)
    rake (13.2.1)
    rb-fsevent (0.11.2)
    rb-inotify (0.11.1)
      ffi (~> 1.0)
    rbtree3 (0.7.1)
    rdoc (6.7.0)
      psych (>= 4.0.0)
    redis-client (0.22.2)
      connection_pool
    reline (0.5.10)
      io-console (~> 0.5)
    rexml (3.3.7)
    ruby2_keywords (0.0.5)
    searchkick (5.4.0)
      activemodel (>= 6.1)
      hashie
    securerandom (0.3.1)
    sidekiq (7.3.2)
      concurrent-ruby (< 2)
      connection_pool (>= 2.3.0)
      logger
      rack (>= 2.2.4)
      redis-client (>= 0.22.2)
    sidekiq-failures (1.0.4)
      sidekiq (>= 4.0.0)
    stringio (3.1.1)
    thor (1.3.2)
    timeout (0.4.1)
    tzinfo (2.0.6)
      concurrent-ruby (~> 1.0)
    useragent (0.16.10)
    webmock (3.24.0)
      addressable (>= 2.8.0)
      crack (>= 0.3.2)
      hashdiff (>= 0.4.0, < 2.0.0)
    webrick (1.8.1)
    websocket-driver (0.7.6)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    zeitwerk (2.6.18)

PLATFORMS
  aarch64-linux
  arm64-darwin-23
  x86_64-linux

DEPENDENCIES
  airbrake (~> 13.0)
  aws-sdk-s3 (~> 1.167)
  bundler (= 2.4.19)
  byebug (~> 11.1)
  doorkeeper (~> 5.7)
  doorkeeper-jwt (~> 0.4)
  elasticsearch (= 7.9.0)
  factory_bot_rails (~> 6.4)
  faker (~> 3.4)
  faraday (~> 1.10)
  input_sanitizer (~> 0.6)
  listen (~> 3.9)
  logstash-event (~> 1.2)
  minitest-spec-rails (~> 7.3)
  parallel (~> 1.26)
  pg (~> 1.5)
  pry-rails (~> 0.3)
  puma (~> 6.4)
  rails (= 7.2.1)
  rexml (~> 3.3)
  searchkick (~> 5.4)
  sidekiq (~> 7.3)
  sidekiq-failures (~> 1.0)
  webmock (~> 3.24)
```

**Comment by deivid-rodriguez** [comment]
Hi! The `require` that's causing the error was removed at https://github.com/rubygems/rubygems/pull/8001, released as RubyGems 3.5.19. Can you upgrade RubyGems (`gem update --system`) and try again? I suspect you may run into some different error, but let's upgrade and see what it is. For now, I'll optimistically close this, but feel free to reopen and post updated backtrace and error messages once you upgrade. Thanks!

**Comment by excelsior** [comment]
Thanks for the reply. You were right, there is a different (although very similar) error this time.

```
LoadError: libruby.so.3.2: cannot open shared object file: No such file or directory - /usr/local/bundle/gems/psych-5.1.2/lib/psych.so
  <internal:/usr/local/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          <internal:/usr/local/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          /usr/local/bundle/gems/psych-5.1.2/lib/psych.rb:13:in `<top (required)>'
          <internal:/usr/local/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          <internal:/usr/local/lib/ruby/site_ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:136:in `require'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems.rb:594:in `load_yaml'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package.rb:555:in `read_checksums'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package.rb:617:in `block (2 levels) in verify'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package/tar_reader.rb:25:in `new'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package.rb:616:in `block in verify'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package/file_source.rb:30:in `open'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package/file_source.rb:30:in `with_read_io'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package.rb:615:in `verify'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/package.rb:596:in `spec'
          /usr/local/lib/ruby/site_ruby/3.3.0/rubygems/installer.rb:291:in `spec'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/rubygems_gem_installer.rb:134:in `spec'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/source/rubygems.rb:186:in `install'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/installer/gem_installer.rb:55:in `install'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/installer/gem_installer.rb:17:in `install_from_spec'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/installer/parallel_installer.rb:133:in `do_install'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/installer/parallel_installer.rb:124:in `block in worker_pool'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/worker.rb:62:in `apply_func'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/worker.rb:57:in `block in process_queue'
          <internal:kernel>:187:in `loop'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/worker.rb:54:in `process_queue'
          /usr/local/lib/ruby/site_ruby/3.3.0/bundler/worker.rb:90:in `block (2 levels) in create_threads'

```

## Environment

```
Bundler       2.5.20
  Platforms   ruby, aarch64-linux
Ruby          3.3.5p100 (2024-09-03 revision ef084cc8f4958c1b6e4ead99136631bef6d8ddba) [aarch64-linux]
  Full Path   /usr/local/bin/ruby
  Config Dir  /usr/local/etc
RubyGems      3.5.20
  Gem Home    /usr/local/bundle
  Gem Path    /root/.local/share/gem/ruby/3.3.0:/usr/local/lib/ruby/gems/3.3.0:/usr/local/bundle
  User Home   /root
  User Path   /root/.local/share/gem/ruby/3.3.0
  Bin Dir     /usr/local/bundle/bin
OpenSSL       
  Compiled    OpenSSL 3.0.14 4 Jun 2024
  Loaded      OpenSSL 3.0.14 4 Jun 2024
  Cert File   /usr/lib/ssl/cert.pem
  Cert Dir    /usr/lib/ssl/certs
Tools         
  Git         2.39.5
  RVM         not installed
  rbenv       not installed
  chruby      not installed
```

## Bundler Build Metadata

```
Built At          2024-09-24
Git SHA           a0fc99594a
Released Version  true
```

## Bundler settings

```
app_config
  Set via BUNDLE_APP_CONFIG: "/usr/local/bundle"
silence_root_warning
  Set via BUNDLE_SILENCE_ROOT_WARNING: true
```

## Gemfile

### Gemfile

```ruby
source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{repo}.git" }

ruby '3.3.5'

gem 'airbrake', '~> 13.0'
gem 'aws-sdk-s3', '~> 1.167'
gem 'bundler', '= 2.5.20'
gem 'doorkeeper', '~> 5.7'
gem 'doorkeeper-jwt', '~> 0.4'
gem 'elasticsearch', '= 7.9.0'
gem 'faraday', '~> 1.10'
gem 'input_sanitizer', '~> 0.6'
gem 'logstash-event', '~> 1.2'
gem 'parallel', '~> 1.26'
gem 'pg', '~> 1.5'
gem 'puma', '~> 6.4'
gem 'rails', '= 7.2.1'
gem 'rexml', '~> 3.3'
gem 'searchkick', '~> 5.4'
gem 'sidekiq', '~> 7.3'
gem 'sidekiq-failures', '~> 1.0'

group :development do
  gem 'listen', '~> 3.9'
  gem 'pry-rails', '~> 0.3'
end

group :development, :test do
  gem 'byebug', '~> 11.1'
  gem 'factory_bot_rails', '~> 6.4'
  gem 'faker', '~> 3.4'
end

group :test do
  gem 'minitest-spec-rails', '~> 7.3'
  gem 'webmock', '~> 3.24'
end
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (7.2.1)
      actionpack (= 7.2.1)
      activesupport (= 7.2.1)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
      zeitwerk (~> 2.6)
    actionmailbox (7.2.1)
      actionpack (= 7.2.1)
      activejob (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      mail (>= 2.8.0)
    actionmailer (7.2.1)
      actionpack (= 7.2.1)
      actionview (= 7.2.1)
      activejob (= 7.2.1)
      activesupport (= 7.2.1)
      mail (>= 2.8.0)
      rails-dom-testing (~> 2.2)
    actionpack (7.2.1)
      actionview (= 7.2.1)
      activesupport (= 7.2.1)
      nokogiri (>= 1.8.5)
      racc
      rack (>= 2.2.4, < 3.2)
      rack-session (>= 1.0.1)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.2)
      rails-html-sanitizer (~> 1.6)
      useragent (~> 0.16)
    actiontext (7.2.1)
      actionpack (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      globalid (>= 0.6.0)
      nokogiri (>= 1.8.5)
    actionview (7.2.1)
      activesupport (= 7.2.1)
      builder (~> 3.1)
      erubi (~> 1.11)
      rails-dom-testing (~> 2.2)
      rails-html-sanitizer (~> 1.6)
    activejob (7.2.1)
      activesupport (= 7.2.1)
      globalid (>= 0.3.6)
    activemodel (7.2.1)
      activesupport (= 7.2.1)
    activerecord (7.2.1)
      activemodel (= 7.2.1)
      activesupport (= 7.2.1)
      timeout (>= 0.4.0)
    activestorage (7.2.1)
      actionpack (= 7.2.1)
      activejob (= 7.2.1)
      activerecord (= 7.2.1)
      activesupport (= 7.2.1)
      marcel (~> 1.0)
    activesupport (7.2.1)
      base64
      bigdecimal
      concurrent-ruby (~> 1.0, >= 1.3.1)
      connection_pool (>= 2.2.5)
      drb
      i18n (>= 1.6, < 2)
      logger (>= 1.4.2)
      minitest (>= 5.1)
      securerandom (>= 0.3)
      tzinfo (~> 2.0, >= 2.0.5)
    addressable (2.8.7)
      public_suffix (>= 2.0.2, < 7.0)
    airbrake (13.0.4)
      airbrake-ruby (~> 6.0)
    airbrake-ruby (6.2.2)
      rbtree3 (~> 0.6)
    aws-eventstream (1.3.0)
    aws-partitions (1.983.0)
    aws-sdk-core (3.209.1)
      aws-eventstream (~> 1, >= 1.3.0)
      aws-partitions (~> 1, >= 1.651.0)
      aws-sigv4 (~> 1.9)
      jmespath (~> 1, >= 1.6.1)
    aws-sdk-kms (1.94.0)
      aws-sdk-core (~> 3, >= 3.207.0)
      aws-sigv4 (~> 1.5)
    aws-sdk-s3 (1.167.0)
      aws-sdk-core (~> 3, >= 3.207.0)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.5)
    aws-sigv4 (1.10.0)
      aws-eventstream (~> 1, >= 1.0.2)
    base64 (0.2.0)
    bigdecimal (3.1.8)
    builder (3.3.0)
    byebug (11.1.3)
    coderay (1.1.3)
    concurrent-ruby (1.3.4)
    connection_pool (2.4.1)
    crack (1.0.0)
      bigdecimal
      rexml
    crass (1.0.6)
    date (3.3.4)
    doorkeeper (5.7.1)
      railties (>= 5)
    doorkeeper-jwt (0.4.2)
      jwt (>= 2.1)
    drb (2.2.1)
    elasticsearch (7.9.0)
      elasticsearch-api (= 7.9.0)
      elasticsearch-transport (= 7.9.0)
    elasticsearch-api (7.9.0)
      multi_json
    elasticsearch-transport (7.9.0)
      faraday (~> 1)
      multi_json
    erubi (1.13.0)
    factory_bot (6.5.0)
      activesupport (>= 5.0.0)
    factory_bot_rails (6.4.3)
      factory_bot (~> 6.4)
      railties (>= 5.0.0)
    faker (3.4.2)
      i18n (>= 1.8.11, < 2)
    faraday (1.10.4)
      faraday-em_http (~> 1.0)
      faraday-em_synchrony (~> 1.0)
      faraday-excon (~> 1.1)
      faraday-httpclient (~> 1.0)
      faraday-multipart (~> 1.0)
      faraday-net_http (~> 1.0)
      faraday-net_http_persistent (~> 1.0)
      faraday-patron (~> 1.0)
      faraday-rack (~> 1.0)
      faraday-retry (~> 1.0)
      ruby2_keywords (>= 0.0.4)
    faraday-em_http (1.0.0)
    faraday-em_synchrony (1.0.0)
    faraday-excon (1.1.0)
    faraday-httpclient (1.0.1)
    faraday-multipart (1.0.4)
      multipart-post (~> 2)
    faraday-net_http (1.0.2)
    faraday-net_http_persistent (1.2.0)
    faraday-patron (1.0.0)
    faraday-rack (1.0.0)
    faraday-retry (1.0.3)
    ffi (1.17.0-aarch64-linux-gnu)
    ffi (1.17.0-arm64-darwin)
    ffi (1.17.0-x86_64-linux-gnu)
    globalid (1.2.1)
      activesupport (>= 6.1)
    hashdiff (1.1.1)
    hashie (5.0.0)
    i18n (1.14.6)
      concurrent-ruby (~> 1.0)
    input_sanitizer (0.6.0)
      activesupport (>= 3.0.0)
      method_struct (>= 0.2.2)
    io-console (0.7.2)
    irb (1.14.0)
      rdoc (>= 4.0.0)
      reline (>= 0.4.2)
    jmespath (1.6.2)
    jwt (2.9.0)
      base64
    listen (3.9.0)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    logger (1.6.1)
    logstash-event (1.2.02)
    loofah (2.22.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.12.0)
    mail (2.8.1)
      mini_mime (>= 0.1.1)
      net-imap
      net-pop
      net-smtp
    marcel (1.0.4)
    method_source (1.1.0)
    method_struct (0.3.0)
    mini_mime (1.1.5)
    minitest (5.25.1)
    minitest-spec-rails (7.3.0)
      minitest (>= 5.0)
      railties (>= 4.1)
    multi_json (1.15.0)
    multipart-post (2.4.1)
    net-imap (0.4.16)
      date
      net-protocol
    net-pop (0.1.2)
      net-protocol
    net-protocol (0.2.2)
      timeout
    net-smtp (0.5.0)
      net-protocol
    nio4r (2.7.3)
    nokogiri (1.16.7-aarch64-linux)
      racc (~> 1.4)
    nokogiri (1.16.7-arm64-darwin)
      racc (~> 1.4)
    nokogiri (1.16.7-x86_64-linux)
      racc (~> 1.4)
    parallel (1.26.3)
    pg (1.5.8)
    pry (0.14.2)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-rails (0.3.11)
      pry (>= 0.13.0)
    psych (5.1.2)
      stringio
    public_suffix (6.0.1)
    puma (6.4.3)
      nio4r (~> 2.0)
    racc (1.8.1)
    rack (3.1.7)
    rack-session (2.0.0)
      rack (>= 3.0.0)
    rack-test (2.1.0)
      rack (>= 1.3)
    rackup (2.1.0)
      rack (>= 3)
      webrick (~> 1.8)
    rails (7.2.1)
      actioncable (= 7.2.1)
      actionmailbox (= 7.2.1)
      actionmailer (= 7.2.1)
      actionpack (= 7.2.1)
      actiontext (= 7.2.1)
      actionview (= 7.2.1)
      activejob (= 7.2.1)
      activemodel (= 7.2.1)
      activerecord (= 7.2.1)
      activestorage (= 7.2.1)
      activesupport (= 7.2.1)
      bundler (>= 1.15.0)
      railties (= 7.2.1)
    rails-dom-testing (2.2.0)
      activesupport (>= 5.0.0)
      minitest
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.6.0)
      loofah (~> 2.21)
      nokogiri (~> 1.14)
    railties (7.2.1)
      actionpack (= 7.2.1)
      activesupport (= 7.2.1)
      irb (~> 1.13)
      rackup (>= 1.0.0)
      rake (>= 12.2)
      thor (~> 1.0, >= 1.2.2)
      zeitwerk (~> 2.6)
    rake (13.2.1)
    rb-fsevent (0.11.2)
    rb-inotify (0.11.1)
      ffi (~> 1.0)
    rbtree3 (0.7.1)
    rdoc (6.7.0)
      psych (>= 4.0.0)
    redis-client (0.22.2)
      connection_pool
    reline (0.5.10)
      io-console (~> 0.5)
    rexml (3.3.7)
    ruby2_keywords (0.0.5)
    searchkick (5.4.0)
      activemodel (>= 6.1)
      hashie
    securerandom (0.3.1)
    sidekiq (7.3.2)
      concurrent-ruby (< 2)
      connection_pool (>= 2.3.0)
      logger
      rack (>= 2.2.4)
      redis-client (>= 0.22.2)
    sidekiq-failures (1.0.4)
      sidekiq (>= 4.0.0)
    stringio (3.1.1)
    thor (1.3.2)
    timeout (0.4.1)
    tzinfo (2.0.6)
      concurrent-ruby (~> 1.0)
    useragent (0.16.10)
    webmock (3.24.0)
      addressable (>= 2.8.0)
      crack (>= 0.3.2)
      hashdiff (>= 0.4.0, < 2.0.0)
    webrick (1.8.1)
    websocket-driver (0.7.6)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.5)
    zeitwerk (2.6.18)

PLATFORMS
  aarch64-linux
  arm64-darwin-23
  x86_64-linux

DEPENDENCIES
  airbrake (~> 13.0)
  aws-sdk-s3 (~> 1.167)
  bundler (= 2.4.19)
  byebug (~> 11.1)
  doorkeeper (~> 5.7)
  doorkeeper-jwt (~> 0.4)
  elasticsearch (= 7.9.0)
  factory_bot_rails (~> 6.4)
  faker (~> 3.4)
  faraday (~> 1.10)
  input_sanitizer (~> 0.6)
  listen (~> 3.9)
  logstash-event (~> 1.2)
  minitest-spec-rails (~> 7.3)
  parallel (~> 1.26)
  pg (~> 1.5)
  pry-rails (~> 0.3)
  puma (~> 6.4)
  rails (= 7.2.1)
  rexml (~> 3.3)
  searchkick (~> 5.4)
  sidekiq (~> 7.3)
  sidekiq-failures (~> 1.0)
  webmock (~> 3.24)
```

This is how Bundler's output looks like before the error occurs: 

```
Install missing gems with `bundle install`
Fetching gem metadata from https://rubygems.org/........
Resolving dependencies...
Ignoring psych-5.1.2 because its extensions are not built. Try: gem pristine psych --version 5.1.2
Ignoring stringio-3.1.1 because its extensions are not built. Try: gem pristine stringio --version 3.1.1
Ignoring stringio-3.1.0 because its extensions are not built. Try: gem pristine stringio --version 3.1.0
```

**Comment by deivid-rodriguez** [comment]
Hi! Right, yeah I'm working on a better error for this at https://github.com/rubygems/rubygems/pull/8054, but other than that this is not a Bundler or RubyGems issue. Essentially psych.so was originally compiled for Ruby 3.2, but now it's being used by Ruby 3.3. That doesn't work. We're recently getting lots of issues like this, and I wonder how people are ending up in this situation more often now 🤔. If you can't share more information about how you got here, we can try figure it out.

**Comment by deivid-rodriguez** [comment]
Well, I finally got to the bottom of this and yeah, we can actually fix this error ourselves. #8104 should fix this!

**Comment by Aqualon** [comment]
I got the same error during `bundle install` inside docker (using rubygems 3.5.21, bundler 2.5.21)
```
LoadError: /app/.bundle/ruby/3.3.0/gems/psych-5.1.2/lib/psych.so: cannot open shared object file: No such file or directory - /app/.bundle/ruby/3.3.0/gems/psych-5.1.2/lib/psych.so
```
but the mentioned file `/app/.bundle/ruby/3.3.0/gems/psych-5.1.2/lib/psych.so` actually exists.

I then just deleted the `psych.so` and got the complaint about `/app/.bundle/gems/stringio-3.1.1/lib/stringio.so` which also existed. I also deleted that file and afterwards `bundle install` worked fine.

The app I tried to run still uses an older psych in it's `Gemfile.lock` 
```
# bundle info psych
  * psych (3.3.4)
	Summary: Psych is a YAML parser and emitter
	Homepage: https://github.com/ruby/psych
	Path: /app/.bundle/ruby/3.3.0/gems/psych-3.3.4
	Reverse Dependencies: 
		webdriver-user-agent (7.9) depends on psych (>= 0)
```
For `stringio` I get this
```
# bundle info stringio
  * stringio (3.1.1)
	Summary: Pseudo IO on String
	Homepage: https://github.com/ruby/stringio
	Path: /usr/local/lib/ruby/gems/3.3.0/gems/stringio-3.1.1
	Default Gem: yes
```
so same version, but now used from the default gems directory of the install ruby 3.3.5.

Not sure if this is also related to https://github.com/rubygems/rubygems/pull/8104 or yet another issue.

**Comment by deivid-rodriguez** [comment]
Yep, that issue will be fixed after #8104 👍.
